mathics/builtin/arithmetic.py

```diff
@@ -571,21 +599,35 @@ def apply(self, items, evaluation):
                     'Plus', Integer(1), leaves[-1].leaves[1]))
             else:
                 leaves.append(item)
-        if number == (1, 0):
+
+        if numbers:
+            if prec is not None:
+                if is_machine_precision:
+                    numbers = [item.to_mpmath() for item in numbers]
+                    number = mpmath.fprod(numbers)
+                    number = Number.from_mpmath(number)
+                else:
+                    with mpmath.workprec(prec):
+                        numbers = [item.to_mpmath() for item in numbers]
+                        number = mpmath.fprod(numbers)
+                        number = Number.from_mpmath(number, dps(prec))
+            else:
+                number = sympy.Mul(*[item.to_sympy() for item in numbers])
+                number = from_sympy(number)
+        else:
+            number = Integer(1)
```

Comment:
Probably. I tried a couple of times to clean this up / subclass `_MPMathFunction` but it just got a bit messy.